lib/ansible/modules/windows/win_mssql_query.ps1

```diff
@@ -0,0 +1,72 @@
+#!powershell
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Daniele Lazzari  <lazzari@mailup.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$ErrorActionPreference = "Stop"
+$result = @{
+    'changed' = $False;
+    'output'  = @()
+}
+Function Invoke-Query {
+    Param(
+        [string]$Query,
+        [int]$QueryTimeout,
+        [string]$Database,
+        [string]$ServerInstance,
+        [string]$ServerInstanceUser,
+        [Security.SecureString]$ServerInstancePassword,
+        [bool]$CheckMode
+    )
+    $output = @()
+    try {
+        if (!($CheckMode)) {
+            if (!($ServerInstanceUser) -or !($ServerInstancePassword)) {
```

Comment:
Consider using splatting here instead;

```
$sql_params = @{
    ServerInstance = $ServerInstance
    Database = $Database
    Query = $Query
    QueryTimeout = $QueryTimeout
}
if ($ServerInstanceUser -and $ServerInstancePassword) {
    $SQLCredential = New-Object TypeName System.Management.Automation.PSCredential ($ServerInstanceUser, $ServerInstancePassword)
    $sql_params["Credential"] = $SQLCredential
}

Invoke-SqlCmd @sql_params
```

The benefit to this is you only call `Invoke-SqlCmd` in one spot, also it's a lot easier to deal with if statements when you aren't checking the false result.

Reply:
Excellent, I didn't know it was possible, I was looking for a solution like this and I'll definitely do it.